src/PowerShell/PowerShellManagerPool.cs

```diff
@@ -15,37 +20,80 @@ namespace Microsoft.Azure.Functions.PowerShellWorker.PowerShell
     /// </summary>
     internal class PowerShellManagerPool
     {
-        private readonly ILogger _logger;
-        // Today we don't really support the in-proc concurrency. We just hold an instance of PowerShellManager in this field.
-        private PowerShellManager _psManager;
+        private readonly int _upperBound = 25;
+        private readonly MessagingStream _msgStream;
+        private readonly AsyncConcurrentPool<PowerShellManager> _pool;
+        private int _poolSize;
 
         /// <summary>
         /// Constructor of the pool.
         /// </summary>
-        internal PowerShellManagerPool(ILogger logger)
+        internal PowerShellManagerPool(MessagingStream msgStream)
         {
-            _logger = logger;
+            string upperBound = Environment.GetEnvironmentVariable("InProcConcurrencyUpperBound");

```

Comment:
Should we maybe use something that includes the word "Worker" or "PowerShell" just so people know where this is actually being used?